### PR TITLE
add `!yolo` to list open pull requests in duckbot

### PR DESCRIPTION
**Pull request by twentylemon**
##### Summary 

Related to #39.

This is part 1 of the issue, allowing the bot owner and repo admins to do `!yolo` to get a list of the open pull requests. It includes links to the pull requests as well as checks status. Part 2 will eventually allow people to merge pull requests, but it's not there yet. For results, check the #duckbutt channel in the beta server.

FYI @Chippers255 the token being used in production is your github access token, the same one used to write to the wiki on push.

Part of this change is some minor cleanup in the weather class and tests. I changed the openweather client method to be a `@property`, and changed the name of a local `clazz` fixture to be `weather`. Intellij was confused when I had two `clazz` fixtures (I made one in the tests here at first).  
Also a minor performance improvement to the `skip_if_private_channel` fixture, which now doesn't create other channels and instead uses the `channel.type` property.

##### Checklist

* [x] this is a source code change
  * [x] run `isort . && black .` in the repository root
  * [x] run `pytest` in the repository root
  * [x] link to issue being fixed using a [_fixes keyword_](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue), if such an issue exists
  * [x] update the wiki documentation if necessary
* [ ] or, this is **not** a source code change

### CODE REVIEW

File: duckbot/cogs/github/yolo_merge.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+from discord.ext import commands
+from github.PullRequest import PullRequest
+
+
+async def is_repository_admin(context: commands.Context):
+    """Disallow !yolo to be used outside of a server, and only allow the bot owner or
+    repository owners to use it."""
+    if context.guild is None:
+        raise commands.NoPrivateMessage()
+    if not await context.bot.is_owner(context.author) and context.author.id not in [368038054558171141, 776607982472921088, 375024417358479380]:
```
Comment (jameshughes89) [inline duckbot/cogs/github/yolo_merge.py:15]: Sick

File: duckbot/cogs/github/yolo_merge.py [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+        await self.list(context)
+
+    async def list(self, context: commands.Context):
+        repo = self.github.get_repo("duck-dynasty/duckbot")
+        pulls = list(repo.get_pulls()[:6])
+        if pulls:
+            embed = self.as_embed(pulls)
+            await context.send(embed=embed)
+        else:
+            await context.send("There's no open pull requests, brother. Never forget to wumbo.")
```
Comment (jameshughes89) [inline duckbot/cogs/github/yolo_merge.py:47]: Wumbo?

File: wiki/Commands.md [hunk 3]
```diff
@@ -74,3 +75,7 @@ DuckBot will roll Dungeons and Dragons style dice for you. See [dice syntax](htt
 > Human: !roll 1d20  
 > DuckBot: Rolls: 1d20 (3)  
 > Total: 3
+
+Yolo Pull Requests
+------------------
+DuckBot will list the open pull requests when the bot owner or one of the repository owners uses the `!yolo` command. DuckBot will eventually learn how to merge his own pull requests, or something.
```
Comment (jameshughes89) [inline wiki/Commands.md:81]: Prog

**Review by jameshughes89 — APPROVED** [review]

File: duckbot/cogs/github/yolo_merge.py (hunk 2 above)
Comment (Chippers255) [reply duckbot/cogs/github/yolo_merge.py:47]: The opposite of mini. You know, I wumbo, you wumbo...

Comment (twentylemon) [reply duckbot/cogs/github/yolo_merge.py:47]: Wumbology, the study of wumbo! Come on James, it's first grade stuff.

**Pull request merged**